verification/verify.go

```diff
@@ -103,7 +112,29 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		return nil, nil
 	}
 
-	// TODO sig checking; extract from broadcast.go
+	if params.Results != nil {
```

Comment:
I think `params.Results` should always be non-nil if B is verifying results from O (otherwise we wouldn't have any results from O) so I don't think the nil check is needed here.

Reply:
I think the `nil` check is indeed needed, or something else may be wrong.

After applying the following:
```diff
diff --git a/verification/verify.go b/verification/verify.go
index a05f39f7..29f872d4 100644
--- a/verification/verify.go
+++ b/verification/verify.go
@@ -112,7 +112,6 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
                return nil, nil
        }
 
-       if params.Results != nil {
        segHashes := make([][]byte, len(params.Results.Segments))
        for i, segment := range params.Results.Segments {
 
@@ -134,7 +133,6 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
                glog.Error("Sig check failed")
                return nil, errPMCheckFailed
        }
-       }
 
        var err error
        res := &Results{}
```
The tests blow up on a nil pointer dereference:
```
go-livepeer/verification$ go test
E0211 15:20:26.387266    1205 epic.go:151] Could not submit request Post : unsupported protocol scheme ""
--- FAIL: TestVerify (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xcdf2b0]

goroutine 114 [running]:
testing.tRunner.func1(0xc00011a600)
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:874 +0x3a3
panic(0x161e660, 0x232ada0)
	/home/mk/.gvm/gos/go1.13.7/src/runtime/panic.go:679 +0x1b2
github.com/livepeer/go-livepeer/verification.(*SegmentVerifier).Verify(0xc000077e98, 0xc0001f8070, 0x0, 0x0, 0x0)
	/home/mk/git/livepeer/go-livepeer/verification/verify.go:115 +0x50
github.com/livepeer/go-livepeer/verification.TestVerify(0xc00011a600)
	/home/mk/git/livepeer/go-livepeer/verification/verify_test.go:39 +0x2ad
testing.tRunner(0xc00011a600, 0x17bcfe8)
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/home/mk/.gvm/gos/go1.13.7/src/testing/testing.go:960 +0x350
exit status 2
FAIL	github.com/livepeer/go-livepeer/verification	0.071s
```

Reply:
Ah I see we have a few tests that don't include a `Results` in the `Params` passed in. Ok.